src/NuGet.Core/NuGet.Protocol/Resources/PackageUpdateResource.cs

```diff
@@ -242,27 +261,54 @@ public Uri SourceUri
                     packagePath));
             }
 
-            var sourceUri = UriUtility.CreateSourceUri(source);
+            Uri packageSourceUri = UriUtility.CreateSourceUri(source);
 
-            if (string.IsNullOrEmpty(apiKey) && !sourceUri.IsFile)
+            if (string.IsNullOrEmpty(apiKey) && !packageSourceUri.IsFile)
             {
                 log.LogWarning(string.Format(CultureInfo.CurrentCulture,
                     Strings.NoApiKeyFound,
                     GetSourceDisplayName(source)));
             }
 
-            await PushAll(source, apiKey, noServiceEndpoint, skipDuplicate, requestTimeout, log, packagesToPush: nupkgsToPush, token);
-        }
-
-        private async Task PushAll(string source, string apiKey, bool noServiceEndpoint, bool skipDuplicate, TimeSpan requestTimeout, ILogger log, IEnumerable<string> packagesToPush, CancellationToken token)
-        {
-            foreach (var packageToPush in packagesToPush)
+            foreach (string nupkgToPush in nupkgsToPush)
             {
-                await PushPackageCore(source, apiKey, packageToPush, noServiceEndpoint, skipDuplicate, requestTimeout, log, token);
+                bool packageWasPushed = await PushPackageCore(source, apiKey, nupkgToPush, noServiceEndpoint, skipDuplicate, requestTimeout, log, token);
+
+                // Push corresponding symbols, if successful.
+                if (packageWasPushed && !string.IsNullOrEmpty(symbolSource))
+                {
+                    bool isSymbolEndpointSnupkgCapable = symbolPackageUpdateResource != null;
+                    string symbolPackagePath = GetSymbolsPath(nupkgToPush, isSnupkg: isSymbolEndpointSnupkgCapable);
+
+                    // There may not be a snupkg with the same filename. Ignore it since this isn't an explicit snupkg push.
+                    if (!File.Exists(symbolPackagePath))
```

Comment:
This is new logic since I'm now "calculating" snupkg filenames. Before it was implicit because this only pushed filenames that were returned by the `LocalFolderUtility.ResolvePackageFromPath`.